Assets/Mirror/Runtime/NetworkClient.cs

```diff
@@ -36,13 +37,25 @@ internal void SetHandlers(NetworkConnection conn)
         public NetworkClient()
         {
             if (LogFilter.Debug) { Debug.Log("Client created version " + Version.Current); }
-            AddClient(this);
+
+            if (singleton != null)
+            {
+                Debug.LogError("NetworkClient: can only create one!");
+                return;
+            }
+            singleton = this;
         }
 
         public NetworkClient(NetworkConnection conn)
         {
             if (LogFilter.Debug) { Debug.Log("Client created version " + Version.Current); }
-            AddClient(this);
+
```

Comment:
call the other constructor instead of duplicating singleton initialization

```cs
public NetworkClient(NetworkConnection conn) : this()
{
    ...
}
```